crates/dekaf/src/session.rs

```diff
@@ -454,16 +454,33 @@ impl Session {
                     }
                 };
 
-                if matches!(self.reads.get(&key), Some(pending) if pending.offset == fetch_offset) {
-                    metrics::counter!(
-                        "dekaf_fetch_requests",
-                        "topic_name" => key.0.to_string(),
-                        "partition_index" => key.1.to_string(),
-                        "state" => "read_pending"
-                    )
-                    .increment(1);
-                    continue; // Common case: fetch is at the pending offset.
+                match self.reads.get(&key) {
+                    Some((_, started_at))
+                        if started_at.elapsed() > std::time::Duration::from_secs(60 * 5) =>
+                    {
+                        metrics::counter!(
+                            "dekaf_fetch_requests",
+                            "topic_name" => key.0.to_string(),
+                            "partition_index" => key.1.to_string(),
+                            "state" => "read_expired"
+                        )
+                        .increment(1);
+                        tracing::debug!(lifetime=?started_at.elapsed(), topic_name=?key.0,partition_index=?key.1, "Restarting expired Read");
+                        self.reads.remove(&key);
+                    }
+                    Some(_) => {
+                        metrics::counter!(
+                            "dekaf_fetch_requests",
+                            "topic_name" => key.0.to_string(),
+                            "partition_index" => key.1.to_string(),
+                            "state" => "read_pending"
+                        )
+                        .increment(1);
+                        continue; // Common case: fetch is at the pending offset.
+                    }
+                    _ => {}
                 }
+
```

Comment:
what's the mechanism by which this session will grab an entirely new gazette::Router (e.g. new underlying connection) for the next started read?

Reply:
I was thinking that the call to `authenticated_client()` always created a new gazette client, but it turns out it only does when refreshing the access token. I updated this to explicitly call `client.with_fresh_gazette_client()` upon timeout. We could also go nuclear and have `authenticated_client()` _always_ return a client with fresh router, but that seemed like it might be unnecessary. Thoughts?